### PR TITLE
Development

### PR DESCRIPTION
**Pull request by BruceJi7**
The current development state is stable, and much better than the current main branch. It is ready to deploy

### CODE REVIEW

File: src/components/SmartContracts/ContractSection/ContractSection.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { State, ContractType } from "../../../utility/types"
+
+import "./ContractSection.css"
+
+const ContractSection = () => {
+  const contracts = useSelector((state: State) => state.ctx.contractData)
+
+  let content = null
+
+  if (contracts.length > 0) {
```
Comment (ivanms1) [inline src/components/SmartContracts/ContractSection/ContractSection.tsx:12]:
```suggestion
  if (contracts.length) {
```

File: src/components/SmartContracts/ContractSection/ContractSection.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      )
+    })
+  }
+
+  return (
+    <section className="section-block section-single">
+      <div className="section-title">Smart Contracts</div>
+      <div className="info-table contracts-header">
+        <div className="table-header-cell">Name</div>
+        <div></div>
```
Comment (ivanms1) [inline src/components/SmartContracts/ContractSection/ContractSection.tsx:35]: ?

File: src/components/SmartContracts/ContractSection/ContractSection.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        <div className="table-header-cell">Versions</div>
+      </div>
+      {content}
```
Comment (ivanms1) [inline src/components/SmartContracts/ContractSection/ContractSection.tsx:38]: remove please, it's just null if it gets here

File: src/store/actions/basicActions.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+const assembleChannelStats = (data: ObjectType) => {
+  return { type: SET_CHANNEL_STATS, payload: { channelStats: data } }
+}
+
+export const getAvailableChannels = () => {
+  return (dispatch: any) => {
+    axios
+      .get("/channels/info")
+      .then((response) => {
+        // console.log(
```
Comment (ivanms1) [inline src/store/actions/basicActions.ts:42]: remove please

File: src/store/actions/blockActions.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+// Redux Action to get block activity data
+export const setBlockActivityData = (
+  channelHash: string,
+  from: number | string | null = null
+) => {
+  return (dispatch: any) => {
+    let axiosAddress = `/blockActivity/${channelHash}`
+    if (from) {
+      // console.log("Block Activity: Using Blocknum: ", from)
```
Comment (ivanms1) [inline src/store/actions/blockActions.ts:32]: remove please

File: src/store/actions/blockActions.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      axiosAddress += `?blocknum=${from}`
+    }
+
+    axios
+      .get(axiosAddress)
+      .then((response) => {
+        // console.log("Block Activity: ", response.status, response.statusText)
```
Comment (ivanms1) [inline src/store/actions/blockActions.ts:39]: remove please

File: src/store/actions/contractActions.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+const assembleAllContracts = (data: Array<ContractType>) => {
+  return { type: GET_ALL_CONTRACTS, payload: { contractData: data } }
+}
+
+// Here, hash is an argument given when this action is called.
+//It is called with this function, since it has extra processes to do first.
+export const getAllContracts = (channelHash: string) => {
+  return (dispatch: any) => {
+    axios.get(`/chaincode/${channelHash}`).then((response) => {
+      console.log("TEST CONTRACT GET: ", response.data.chaincode)
```
Comment (ivanms1) [inline src/store/actions/contractActions.ts:17]: remove please

File: src/store/actions/txnActions.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+// Redux Action to get transaction activity data
+export const setTxnActivityData = (
+  channelHash: string,
+  from: number | string | null = null
+) => {
+  return (dispatch: any) => {
+    let axiosAddress = `/txActivity/${channelHash}`
+    if (from) {
+      // console.log("Transaction Activity: Using TX ID: ", from)
```
Comment (ivanms1) [inline src/store/actions/txnActions.ts:29]: remove please

File: src/store/actions/txnActions.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      axiosAddress += `?txId=${from}`
+    }
+
+    axios
+      .get(axiosAddress)
+      .then((response) => {
+        // console.log(
```
Comment (ivanms1) [inline src/store/actions/txnActions.ts:36]: remove please

File: src/store/actions/contractActions.ts (hunk 7 above)
Comment (BruceJi7) [reply src/store/actions/contractActions.ts:17]: This console.log is still in use for developing that page, I will keep this for now

**Review by ivanms1 — APPROVED** [review]

**Pull request merged**